test/Unit/Durable/DurableTimerTests.cs

```diff
@@ -42,43 +42,101 @@ public void CreateTimerOrContinue_WaitsUntilTimerFires(bool timerScheduledAndFir
+                _durableTimer.StopAndCreateTimerOrContinue(context:context, fireAt: _fireAt, noWait: false, _ => { Assert.True(false, "Unexpected output"); });
                 Assert.Equal(_restartTime, context.CurrentUtcDateTime);
             }
+            VerifyCreateDurableTimerActionAdded(context, _fireAt);
         }
 
-        private HistoryEvent[] CreateTimerFiredHistory(bool timerScheduledAndFired, DateTime fireAt, DateTime restartTime, bool orchestratorStartedIsProcessed)
+        [Fact]
+        public void StopAndCreateTimerOrContinue_ContinuesToNextTimer_IfTimersHaveIdenticalFireAt()
         {
-            var history = new List<HistoryEvent>();
+            var history = DurableTestUtilities.MergeHistories(
+                CreateOrchestratorStartedHistory(date: _startTime, isProcessed: true),
+                CreateDurableTimerHistory(timerCreated: true, timerFired: true, fireAt: _fireAt, restartTime: _restartTime, orchestratorStartedIsProcessed: false),
+                CreateDurableTimerHistory(timerCreated: true, timerFired: true, fireAt: _fireAt, restartTime: _restartTime, orchestratorStartedIsProcessed: false),
+                CreateDurableTimerHistory(timerCreated: true, timerFired: true, fireAt: _fireAt, restartTime: _shouldNotHitTime, orchestratorStartedIsProcessed: false)
+            );
+
+            var context = new OrchestrationContext { History = history };
+
+            for (int i = 0; i < 2; i++) {
+                _durableTimer.StopAndCreateTimerOrContinue(context: context, fireAt: _fireAt, noWait: false, _ => { Assert.True(false, "Unexpected output"); });
+                Assert.Equal(_restartTime, context.CurrentUtcDateTime);
+            }
+
+            _durableTimer.StopAndCreateTimerOrContinue(context: context, fireAt: _fireAt, noWait: false, _ => { Assert.True(false, "Unexpected output"); });
+            Assert.Equal(_shouldNotHitTime, context.CurrentUtcDateTime);
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void StopAndCreateTimerOrContinue_OutputsNothing_IfNoWaitNotRequested(bool timerCreated, bool timerFired)
+        {
+            var history = DurableTestUtilities.MergeHistories(
+                CreateOrchestratorStartedHistory(date: _startTime, isProcessed: true),
+                CreateDurableTimerHistory(timerCreated: timerCreated, timerFired: timerFired, fireAt: _fireAt, restartTime: _restartTime, orchestratorStartedIsProcessed: false)
+            );
+
+            var context = new OrchestrationContext { History = history };
 
-            if (timerScheduledAndFired)
+            if (!timerCreated)
             {
+                EmulateStop(_durableTimer);
+            }
+
+            _durableTimer.StopAndCreateTimerOrContinue(context: context, fireAt: _fireAt, noWait: false, _ => { Assert.True(false, "Unexpected output"); } );
```

Comment:
Perhaps this is the only test that must have `Assert.True(false, "Unexpected output")`. Consider removing this assert from other tests, just pass `_ => { }`. First of all, this assert will not distract the reader of other tests. Also, if we break the production code and it suddenly start returning something when -NoWait is absent, we will have exactly one unit test failing, and the name of this failing test will point out the reason.